Geometry_Engine/Compute/GrahamScan.cs

```diff
@@ -0,0 +1,141 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.Engine.Base;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Implements the GrahamScan algorithim to determine the convex hull of a list of points.")]
+        [Input("pts", "The points to determine the convex hull.")]
+        [Output("c", "The convex hull of the point list.")]
+        public static List<Point> GrahamScan(List<Point> pts)
+        {
+            if (pts.IsNullOrEmpty())
+            {
+                Base.Compute.RecordError("The point list is null or empty.");
+                return null;
+            }
+
+            // Find the point with the lowest Y coordination
+            IOrderedEnumerable<Point> orderedPts = pts.OrderBy(pt => pt.Y);
+
+            Point p = pts.First();
```

Comment:
It needs to be sorted to find the lowest Y coordinate, so I kept it as an `IOrderedEnumerable<Point> so we can check whether there is more than one point with the lowest Y coordinate. If there are more than one, we can `ThenBy` to sort by X. Otherwise, just to cast `ToList()`.